spec.md

```diff
@@ -2320,139 +2266,52 @@ module DUT :
-    connect out, read(in_ref)
-    node n1 = and(in, UInt<4>(1))
-    node n2 = and(in, UInt<4>(3))
-    define r1 = probe(n1)
-    define r2 = probe(n2)
-```
+## External Modules
 
-### IO with references to endpoint data
+Probe types may be specified as part of an external module (see [@sec:externally-defined-modules]), with the resolved referent for each specified using `ref`{.firrtl} statements.
 
-A primary motivation for input probe references is that in some situations they make it easier to generate the FIRRTL code.
-While output references necessarily capture this design equivalently, this can be harder to generate and so is useful to support.
+## Limitations
 
-The following demonstrates an example of this, where it's convenient to use the same bundle type as both output to one module and input to another, with references populated by both modules targeting signals of interest at each end.
-For this to be the same bundle type -- input on one and output on another -- the `Probe` references for each end should be output-oriented and accordingly are input-oriented at the other end.
-It would be inconvenient to generate this design so that each has output probe references only.
+### `RWProbe`{.firrtl} with Aggregate Types
 
-The `Connect` module instantiates a `Producer` and `Consumer` module, connects them using a bundle with references in both orientations, and forwards those references for inspection up the hierarchy.
-The probe targets are not significant, here they are the same data being sent between the two, as stored in each module.
+Probes of aggregates do not working under all ABI versions.
+In these cases, it is recommended that if you have a need to probe a bundle, you instead use a bundle of probes, each targeting an individual subcomponent.
 
-``` firrtl
-module Consumer:
-  input in : {a: UInt, pref: Probe<UInt>, flip cref: Probe<UInt>}
-  ; ...
-  node n = in.a
-  define in.cref = probe(n)
+For more information on the how probes are lowered, see the FIRRTL ABI Specification.
 
-module Producer:
-  output out : {a: UInt, pref: Probe<UInt>, flip cref: Probe<UInt>}
-  wire x : UInt
-  define out.pref = probe(x)
-  ; ...
-  connect out.a, x
+### No `RWProbe`{.firrtl}s of Ports on Public Modules
 
-module Connect:
-  output out : {pref: Probe<UInt>, cref: Probe<UInt>}
+Conceptually, in order to force a value onto a circuit component, you need to be able to disconnect it from its former driver.
+However, the driver of a port is another port in some other module.
+And so, to force a port, you must have control over both "ends" of a port.
 
-  inst a of Consumer
-  inst b of Producer
+Trouble arises with ports on a public module ([@sec:public-modules]).
+Since we don't control the remote side of ports (since they might live in another FIRRTL file), we cannot force these ports either.
+For this reason, it is an error to use `rwprobe`{.firrtl} on any port on a public module.
```

Comment:
Up to you, but at least to me this is same issue for public and extmodule's (and what used to be the main module/top)-- (the module definition, if FIRRTL-provided, must be public for it to be used as extmodule elsewhere).  Anyway I think this section could be tidied a little, I'll revisit with more concrete suggestions or we can work on it together if you'd like.

Reply:
FWIW neither the other side of a public of extmodule necessarily is defined in FIRRTL.